VendingMachine/VendingMachine.Core/RefundCalculator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+
+namespace Vending.Core
+{
+    public class RefundCalculator
+    {
+        public IDictionary<Coin, int> CalculateRefund(int priceInCents, int paidInCents)
+        {
+            /* Note: I'm not particularly fond of the implementation here.
```

Comment:
The implementation is still pretty stateful, but now cleaner. 
I'm honestly unsure of how to make it any better, but am open to ideas.